utils/httpstream.go

```diff
@@ -44,17 +46,45 @@ func StartHTTPServer(handler Handler, path, format string) {
 	}
 
 	// try to remove pre-existing UDS: ignore errors here
-	if tokens[0] == "unix" {
-		os.Remove(tokens[1])
+	phonySock := "/tmp/phonyfn.sock"
+	if uri.Scheme == "unix" {
+		os.Remove(phonySock)
 	}
 
-	listener, err := net.Listen(tokens[0], tokens[1])
+	listener, err := net.Listen(uri.Scheme, phonySock)
 	if err != nil {
-		panic("net.Listen error: " + err.Error())
+		log.Fatalln("net.Listen error: ", err)
+	}
+
+	if uri.Scheme == "unix" {
+		sockPerm(phonySock, uri.Path)
 	}
 
 	err = server.Serve(listener)
 	if err != nil && err != http.ErrServerClosed {
-		panic("server.Serve error: " + err.Error())
+		log.Fatalln("serve error: ", err)
+	}
+}
+
+func sockPerm(phonySock, realSock string) {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	// somehow this is the best way to get a permissioned sock file, don't ask questions, life is sad and meaningless
+	f, err := os.Create(phonySock)
+	if err != nil {
+		log.Fatalln("error creating sock file", err)
+	}
+
+	err = f.Chmod(0666)
+	if err != nil {
+		f.Close()
+		log.Fatalln("error giving sock file a perm", err)
+	}
+	f.Close()
+
+	err = os.Symlink(realSock, phonySock)
```

Comment:
Is Symlink right here? Won't this create a symlink that is out of the tree of the agent?  (i.e. agent will interpret this symlink as a link to it's own tmp) 

Wonder if this should be: 
* get dir of FN_LISTENER
* Create socket in random filename in that dir
* chmod file
* link or symlink within that dir

Reply:
that would be the 'right thing' to do... will do

Reply:
https://github.com/fnproject/fdk-go/pull/31/commits/ab0e6b6d00656c072bd474f8f291b99351763e26

Reply:
this made me run into a similar issue to i think what you saw today for java, since fn sends in /iofs and not /tmp... d'oh. need to change this in fn... inc

Reply:
not inc* see your PR